Clients/GUI/Pages/SignUp.razor

```diff
@@ -1,57 +1,74 @@
-            </div>
-            <br />
-            <div class="form-group">
-                <label for="Password">Password</label>
-                <InputText class="form-control col-3" type="password" @bind-Value="userInput.Password" placeholder="*****" />
-            </div>
-            <br />
-            <div class="form-group">
-                <label for="ConfirmPassword">Confirm Password</label>
-                <InputText class="form-control col-3" type="password" @bind-Value="userInput.ConfirmPassword" placeholder="*****" />
-            </div>
-            <br />
-            <ValidationSummary />
-            <br />
-
-            <input type="submit" class="form-control col-1 button btn-primary" value="Sign up" />
-
-        </EditForm>
-    </div>
+<div class="col-12">
+    <h3><b>Sign up</b></h3>
+    <hr />
+
+    <EditForm Model="@userInput" OnValidSubmit="@TrySignUp">
+        <DataAnnotationsValidator />
+        <div class="form-group">
+            <label for="Email">Email</label>
+            <InputText class="form-control col-3" @bind-Value="userInput.Email" placeholder="example@gmail.com" />
+            <label style=“color:red;display:@isErrorDisplayed”> @errorMessage </label>
+        </div>
+        <br />
+        <div class="form-group">
+            <label for="Password">Password</label>
+            <InputText class="form-control col-3" type="password" @bind-Value="userInput.Password" placeholder="*****" />
+        </div>
+        <br />
+        <div class="form-group">
+            <label for="ConfirmPassword">Confirm Password</label>
+            <InputText class="form-control col-3" type="password" @bind-Value="userInput.ConfirmPassword" placeholder="*****" />
+        </div>
+        <br />
+        <ValidationSummary />
+        <br />
+
+        <input type="submit" class="form-control col-1 button btn-primary" value="Sign up" />
+
+    </EditForm>
+</div>
 
 
 @code {
     bool isErrorDisplayed = false;
     string errorMessage;
     private SignUpViewModel userInput;
 
-    public void TrySignUp()
+    public async Task TrySignUp()
     {
         try
         {
-            SignUpper.SignUp(new UserEmailPassword(userInput.Email, userInput.Password));
-            NavigationManager.NavigateTo("/signin");
+            var hash = ShaHash.GetHash(userInput.Password);
+            var credentials = new UserEmailPassword(userInput.Email, hash);
+            SignUpper.SignUp(credentials);
+            await LocalStorage.ClearAsync();
+            var userToken = SignInner.SignIn(credentials);
+            LocalStorage.SetItemAsync(userToken.UserId.ToString(), userToken.Body);
+            AuthenticationProvider.MarkSignedIn();
+            NavigationManager.NavigateTo("/userinfo");
+        }
+        catch (BadRequestException e)
+        {
+            userInput = new SignUpViewModel();
+            isErrorDisplayed = true;
+            errorMessage = "Something went wrong during registration";
```

Comment:
Maybe use a field message instead of "Something went wrong during registration"?

Reply:
> Maybe use a field message of exception instead of "Something went wrong during registration"?

I think user would like this one more than "403 Bad Request" or sth like that

Reply:
I think right now it is better to know where we have an error without debug. This message can be change later

Reply:
> I think right now it is better to know where we have an error without debug. This message can be change later

Why don't you like debug? You should print too much information in GUI anyway or you won't get enough information without the debug to find the exact place where error occurs